src/utils/trust.details.ts

```diff
@@ -147,23 +152,30 @@ export const postTrustDetails = async (req: Request, res: Response, next: NextFu
     }
 
     //  if present, get existing trust from session (as it might have attached trustees)
-    const trust = getTrustByIdFromApp(appData, details.trust_id);
+    const trust = getTrustByIdFromApp(appData, details.trust_id, isReview);
     Object.keys(details).forEach(key => trust[key] = details[key]);
 
     //  update trust  in application data at session
-    appData = saveTrustInApp(appData, trust);
+    if (!isReview) {
+      appData = saveTrustInApp(appData, trust);
+    }
 
     //  update trusts in beneficial owners
     const selectedBoIds = req.body?.beneficialOwnersIds ?? [];
     appData = updateBeneficialOwnersTrustInApp(appData, details.trust_id, selectedBoIds);
 
+    // if reviewing a trust, mark trust as in review
+    if (isReview) {
```

Comment:
Should the trust not be already in review? We may never get the post coming in from the user. Should be marking as having reviewed the trust (and trust links reviewed too)?